### PR TITLE
final project review

### PR DESCRIPTION
**Pull request by maticrown**

### CODE REVIEW

File: src/bloomfilter.py [hunk 1]
```diff
@@ -13,22 +13,26 @@ class BloomFilter(object):
     Class for Bloom filter, using murmur3 hash function 
     '''
   
-    def __init__(self, items_count,fp_prob): 
+    def __init__(self, size,items_count):
```
Comment (EladCh) [inline src/bloomfilter.py:16]: change `items_count` to `times_hash`

File: src/bloomfilter.py [hunk 2]
```diff
@@ -13,22 +13,26 @@ class BloomFilter(object):
...
+    def __init__(self, size,items_count): 
         ''' 
         items_count : int 
             Number of items expected to be stored in bloom filter 
         fp_prob : float 
             False Positive probability in decimal 
         '''
         # False posible probability in decimal 
-        self.fp_prob = fp_prob 
+        #self.fp_prob = fp_prob
```
Comment (EladCh) [inline src/bloomfilter.py:24]: remove unnecessary lines (commented)

File: src/bloomfilter.py [hunk 3]
```diff
@@ -13,22 +13,26 @@ class BloomFilter(object):
...
         fp_prob : float 
             False Positive probability in decimal 
         '''
         # False posible probability in decimal 
-        self.fp_prob = fp_prob 
+        #self.fp_prob = fp_prob 
   
         # Size of bit array to use 
-        self.size = self.get_size(items_count,fp_prob) 
+        #self.size = self.get_size(size,items_count)
```
Comment (EladCh) [inline src/bloomfilter.py:27]: remove unnecessary lines (commented)

File: src/bloomfilter.py [hunk 4]
```diff
@@ -13,22 +13,26 @@ class BloomFilter(object):
...
   
         # Size of bit array to use 
-        self.size = self.get_size(items_count,fp_prob) 
+        #self.size = self.get_size(size,items_count) 
   
+        self.size = size
+
         # number of hash functions to use 
-        self.hash_count = self.get_hash_count(self.size,items_count) 
+        #self.hash_count = self.get_hash_count(self.size,items_count)
```
Comment (EladCh) [inline src/bloomfilter.py:32]: remove unnecessary lines (commented)

File: README.md [hunk 5]
```diff
@@ -1,12 +1,25 @@
...
+$ sudo apt-get install -y python3-pip
+```
+## Dependencies for centos7:
+```
+$ sudo yum update
+$ sudo yum install -y python36-setuptools
+$ sudo yum install -y python3-devel
+$ sudo easy_install-3.6 pip
+
+```
```
Comment (EladCh) [inline README.md:18]: move the "```" from line 18 to line 17

File: src/bloomfilter.py [hunk 6]
```diff
@@ -41,7 +45,14 @@ def __init__(self, items_count,fp_prob):
         # maximum items in the filter (from input)
         self.max_num_of_items = items_count
         
-  
+        self.values = list()
```
Comment (EladCh) [inline src/bloomfilter.py:48]: add a comment above: "# holds the values that entered the filter"

File: src/bloomfilter.py [hunk 7]
```diff
@@ -59,11 +70,27 @@ def add(self, item):
...
+                # if any of bit is False then,its not present 
+                # in filter 
+                # else there is probability that it exist 
+                return False
+        '''
+        if item in self.values:
+            return True
+        return False
+
+    def check(self,item):
```
Comment (EladCh) [inline src/bloomfilter.py:93]: add a description of the function-  '''  Check for existence of an item in filter according to the bitarray  '''

File: src/bloomfilter.py [hunk 8]
```diff
@@ -59,11 +70,27 @@ def add(self, item):
...
         # increment element coount
         self.element_count += 1
+
+        (self.values).append(item)
   
-    def check(self, item): 
+    def check_values(self, item): 
         ''' 
         Check for existence of an item in filter 
         '''
```
Comment (EladCh) [inline src/bloomfilter.py:79]: change the description of the function-  '''  Check for existence of an item in filter according to self.values '''

File: src/bloomfilter.py [hunk 9]
```diff
@@ -59,11 +70,27 @@ def add(self, item):
...
+        '''for i in range(self.hash_count): 
+            digest = mmh3.hash(item,i) % self.size 
+            if self.bit_array[digest] == False: 
+  
+                # if any of bit is False then,its not present 
+                # in filter 
+                # else there is probability that it exist 
+                return False
+        '''
```
Comment (EladCh) [inline src/bloomfilter.py:88]: remove unnecessary comment

File: src/bloomfilter.py [hunk 10]
```diff
@@ -74,6 +101,7 @@ def check(self, item):
                 return False
         return True
 
+
```
Comment (EladCh) [inline src/bloomfilter.py:104]: remove this space line

File: src/gui.py [hunk 11]
```diff
@@ -43,12 +44,18 @@
 
 # 10-million-password-list from https://github.com/danielmiessler/SecLists/blob/master/Passwords/Common-Credentials/10-million-password-list-top-1000000.txt
 # read it once in the begining and than use it to prevent those passwords
-password_corpus = [line.strip() for line in open("files/10-million-password-list-top-1000000.txt", 'r')]
+password_corpus = [line.strip() for line in open("10-million-password-list-top-1000000.txt", 'r')]
```
Comment (EladCh) [inline src/gui.py:47]: why did you remove "files/" in the path?

File: src/gui.py [hunk 12]
```diff
@@ -58,54 +65,69 @@
...
+            # reset redundency variables
+            redundency_count = 0
+            redundency_map = dict()
     elif event == 'Insert new password':
         try:
             # first make sure that the password doesnt apear in the password_corpus
-            if values['-NEW-PASSWORD-'] not in password_corpus:
+            if english_dictionary.check(values['-NEW-PASSWORD-']) == True:
+                sg.PopupError("The password exists in the English standard dictionary, pick another") 
+            if values['-NEW-PASSWORD-'] in password_corpus:
```
Comment (EladCh) [inline src/gui.py:82]: i think elif is better. what happens when you have a word that is in both english_dictionary and password_corpus? there will be 2 popups?

File: src/gui.py [hunk 13]
```diff
@@ -146,10 +168,91 @@
...
+            if in_dict > 0 and in_dict < 3:
+                popup_txt = 'The password is in dictionary ' + str(in_dict)
+                sg.PopupOK(popup_txt)
+            elif in_dict == 3:
+                sg.PopupOK('Th password is in both dictionaries 1 & 2')
+            else:
+                sg.PopupError('The password is not present in any dictionary')
+            
+            # TODO decide what to do with the test of the previously known dictionary (the comented line below)
+            #test_bloom_filter(int(values['-K-']),float(values['-N-']))
```
Comment (EladCh) [inline src/gui.py:193]: you can remove this comment

File: src/gui.py [hunk 14]
```diff
@@ -146,10 +168,91 @@
...
+        try:
+            insert_count = bloomf_1.get_element_count() + bloomf_2.get_element_count()
+            marked_bits_dict1 = bloomf_1.get_marked_bits_count()
+            marked_bits_dict2 = bloomf_2.get_marked_bits_count()
+            sg.PopupOK('Insert Counter: {}\n\nMarked Bits Counter in dict 1: {}\nMarked Bits Counter in dict 2: {}\n\nRedundent Passwords in both dictionaries: {}'.format(insert_count,marked_bits_dict1, marked_bits_dict2, redundency_count))
+        except:
+            pass
+    elif event == '    Show full visual of the filter    ':
+        #dict1_hex = bytearray(bloomf_1.size).hex()
+        #sg.PopupOKCancel(dict1_hex)
```
Comment (EladCh) [inline src/gui.py:206]: you can remove this comment

File: src/gui.py [hunk 15]
```diff
@@ -146,10 +168,91 @@
...
+
+            dict1_bits = str(bloomf_1.get_bit_array().unpack(zero=b'0', one=b'1')).replace('b',"").replace("'","")
+            dict2_bits = str(bloomf_2.get_bit_array().unpack(zero=b'0', one=b'1')).replace('b',"").replace("'","")
+            #dict1_hexa =
+            #dict2_hexa =
+
+            #sg.PopupOKCancel(dict1_hexa)
+
+            #hexa1 = dict1_hexa.decode
+            #hexa2 = dict2_hexa.decode
```
Comment (EladCh) [inline src/gui.py:221]: remove

File: src/gui.py [hunk 16]
```diff
@@ -146,10 +168,91 @@
...
+            pass
+        except:
+            pass
+
+
+main_window.close()
 
-main_window.close()
\ No newline at end of file
+c = str(bloomf_1.get_bit_array().unpack(zero=b'0', one=b'1')).replace('b',"").replace("'","")
```
Comment (EladCh) [inline src/gui.py:258]: what is this?
remove

File: src/password_strength_rules.py [hunk 17]
```diff
@@ -17,47 +23,51 @@ def password_rules(password):
...
         strengths[3][1] = 'True'
     else:
-        popup_msg="Password contains no digits\nPlease check the analysis for correction"
+        popup_msg+="Password contains no digits.\n"
         mandatory_violation = True
 
-    if any(x.isalnum()for x in password):
-        num_of_signs = [x for x in password if x.isalnum()]
+
+###changed to check for special character
```
Comment (EladCh) [inline src/password_strength_rules.py:64]: please indent the comment and leave only one #

File: src/password_strength_rules.py [hunk 18]
```diff
@@ -68,7 +78,95 @@ def password_rules(password):
             score += (len(num_of_signs)*2)
             strengths[5][1] = 'True'
 
-    # TODO penalties reduction
+
+## checking if string is all letters
```
Comment (EladCh) [inline src/password_strength_rules.py:82]: please indent the comment and leave only one #

File: src/password_strength_rules.py [hunk 19]
```diff
@@ -68,7 +78,95 @@ def password_rules(password):
...
+    if (password.isalpha()):
+        score -= (len(password)/4)
+        popup_msg+="Password is all letters.\n"
+        mandatory_violation=True
+        weaknesses[1][1]='True'
+
+## checking if string is all digits
```
Comment (EladCh) [inline src/password_strength_rules.py:89]: please indent the comment and leave only one #

File: src/password_strength_rules.py [hunk 20]
```diff
@@ -68,7 +78,95 @@ def password_rules(password):
...
+    if(password.isdigit()):
+        score -= (len(password)/4)
+        popup_msg+="Password is all digits.\n"
+        mandatory_violation=True
+        weaknesses[2][1]='True'
+
+
+### checking for repeated characters
```
Comment (EladCh) [inline src/password_strength_rules.py:97]: please indent the comment and leave only one #

File: src/password_strength_rules.py [hunk 21]
```diff
@@ -68,7 +78,95 @@ def password_rules(password):
...
+    results=0
+    results = collections.Counter(password)
+    for i in results:
+        if results[i]>1:
+            score -= results[i]*(results[i]-1)
+            popup_msg+="Password has repeated characters.\n"
+            mandatory_violation=True
+            weaknesses[3][1]='True'
+
+### checking if there are consecutive uppercase characters
```
Comment (EladCh) [inline src/password_strength_rules.py:107]: please indent the comment and leave only one #

File: src/password_strength_rules.py [hunk 22]
```diff
@@ -68,7 +78,95 @@ def password_rules(password):
...
+        else:
+            countUpper=1
+
+        if countUpper>1:
+            score -= countUpper*2
+            popup_msg+="Password has consecutive uppercase letters.\n"
+            mandatory_violation=True
+            weaknesses[4][1]='True'
+
+### checking if there are consecutive lowercase characters
```
Comment (EladCh) [inline src/password_strength_rules.py:123]: please indent the comment and leave only one #

File: src/password_strength_rules.py [hunk 23]
```diff
@@ -68,7 +78,95 @@ def password_rules(password):
...
+        if(letter is not 0):
+            if ord(password[letter])==ord(password[letter-1])-1 and letter.isalpha():
+                sequentialLettersCountDown+=1
+    if sequentialLettersCountUp>=4 or sequentialLettersCountDown>=4:
+        score -= (sequentialLettersCountUp+sequentialLettersCountDown)
+        popup_msg+="Password has sequential letters.\n"
+        mandatory_violation=True
+        weaknesses[6][1]='True'
+
+##checking if there are sequential digits
```
Comment (EladCh) [inline src/password_strength_rules.py:153]: please indent the comment and leave only one #

File: src/password_strength_rules.py [hunk 24]
```diff
@@ -81,4 +179,8 @@ def password_rules(password):
         score_txt = 'Strong'
     elif score >= 80:
         score_txt = 'Very Strong'
-    return weaknesses, strengths, score_txt, popup_msg, mandatory_violation
+
+    print(weaknesses)
+    print(strengths)
```
Comment (EladCh) [inline src/password_strength_rules.py:184]: is it really necessary? if not please remove

File: src/gui.py (hunk 11 above)
Comment (maticrown) [reply src/gui.py:47]: Because for me "files" didn't exist and the code failed until I referenced the document this way

Comment (EladCh) [reply src/gui.py:47]: ok. make sure you write in the installation guide the exact expected location for the txt files
